nasa-jpl/explorer-1
# Extend BlockTable styles to support new content types

### Checklist

  - PR titles are used to generate the change log in [releases](../releases); good ones make that easier to scan.
  - PRs will be broadly categorized in the change log, but for even easier scanning, consider prefixing with a component name or other useful categorization, e.g., "BaseButton: fix layout bug", or "Storybook: Update dependencies".
  - These tags are used to do the aforementioned broad categorization in the change log and determine what the next release's version number should be.
  - [Release Drafter](https://github.com/marketplace/actions/release-drafter) will attempt to do the category labeling for you! Please double-check its work.

## Description

These styles specifically pair with the typed table block template in WCP PR 594. Some of them could potentially be applied as inline Tailwind classes, but I've kept them all here, at least for now for easier reviewing.

It may be better to create a mixin for some of the styles we want to get from BlockText, rather than using `@extend`, which it doesn't appear we've ever done before.

## Instructions to test

**Note:** This PR is branched off of the `1.0.0-beta.2` tag in this repo to make sure it could be tested with the current version of the package used in WCP.

In WCP, on the `features/typed-table-block` branch from PR 594:
1. `make start`
1. Create a test page and put a table block on it with some test content
1. Load up your test page in the browser and see an unstyled table

In another shell:
1. `cd cms/theme/static_src/`
1. `npm link @nasa-jpl/explorer-1`
1. Ignore the failing build in the cms container output, which was triggered by watch
1. `rm -rf .parcel-cache && npx parcel build ./src/index.js --dist-dir ../static/ --no-source-maps` (which is `npm run buld:all` but building the CSS and JS only, leaving out the fonts, images, etc.)
1. Reload your test page (because Browsersync doesn’t trigger after the failed build in the container)
1. See the styles you added in your local explorer-1 repo!

Watch won't work, so re-run the `rm -rf .parcel-cache && npx parcel build ./src/index.js --dist-dir ../static/ --no-source-maps` if you want to see any changes you make to the SCSS in explorer-1

## Tested in the following environments/browsers:



### Operating System


### Browser
